runner/src/server/plugins/engine/plugin.ts

```diff
@@ -88,27 +89,59 @@ export const plugin = {
             basePath: id,
           });
           return h.response({}).code(204);
-        },
-      });
+        } else {
+          console.log(
+            "\x1b[36m%s\x1b[0m",
+            `**********
+NOT FOUND: The /publish route is not accessible as config.previewMode is disabled. 
+To enable it, please switch to development mode by setting NODE_ENV=development
+in your environment, or create a custom environment config with PREVIEW_MODE=true
+and ENFORCE_CSRF=false. See runner/config/development.json for env defaults
+**********`
+          );
+          throw Boom.notFound(
+            `previewMode is disabled. To enable it, please switch to development mode 
+by setting NODE_ENV=development in your environment`
+          );
+        }
+      },
+    });
 
-      server.route({
-        method: "get",
-        path: "/published/{id}",
-        handler: (request: HapiRequest, h: HapiResponseToolkit) => {
+    server.route({
+      method: "get",
+      path: "/published/{id}",
+      handler: (request: HapiRequest, h: HapiResponseToolkit) => {
+        if (previewMode) {
           const { id } = request.params;
           if (forms[id]) {
             const { values } = forms[id];
             return h.response(JSON.stringify({ id, values })).code(200);
           } else {
             return h.response({}).code(204);
           }
-        },
-      });
+        } else {
+          console.log(
+            "\x1b[36m%s\x1b[0m",
```

Comment:
p.s. I was using https://www.npmjs.com/package/cli-ux for displaying warnings to the user before starting the app (but it's since been deprecated/replaced)